test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/RevIncludeSearchTests.cs

```diff
@@ -0,0 +1,228 @@
+            Assert.Equal(1, bundle.Total);
+
+            // ensure that the included resources are not counted when _total is specified and the results fit in a single bundle.
+            bundle = await Client.SearchAsync(ResourceType.Organization, $"{query}&_total=accurate");
+            Assert.Equal(1, bundle.Total);
+        }
+
+        [Fact]
+        public async Task GivenARevIncludeSearchExpression_WhenSearchedWithPost_ThenCorrectBundleShouldBeReturned()
+        {
+            // Delete all locations before starting the test.
+            await Client.DeleteAllResources(ResourceType.Location);
+            var organizationResponse = await Client.CreateAsync(new Organization());
+
+            var locationResponse = await Client.CreateAsync(new Location
+            {
+                ManagingOrganization = new ResourceReference($"Organization/{organizationResponse.Resource.Id}"),
+            });
+
+            Bundle bundle = await Client.SearchPostAsync(ResourceType.Organization.ToString(), default, ("_revinclude", "Location:managingOrganization"));
+
+            ValidateBundle(
+                bundle,
+                locationResponse.Resource,
+                organizationResponse.Resource);
+
+            ValidateSearchEntryMode(bundle, ResourceType.Organization);
+        }
+
+        [Fact]
+        public async Task GivenARevIncludeSearchExpressionWithSimpleSearch_WhenSearched_ThenCorrectBundleShouldBeReturned()
+        {
+            string query = $"_tag={Fixture.Tag}&_revinclude=Location:managingOrganization&code=429858000";
+
+            Bundle bundle = await Client.SearchAsync(ResourceType.Organization, query);
+
+            /* ValidateBundle(
+                bundle,
+                Fixture.SmithSnomedDiagnosticReport,
+                Fixture.SmithPatient,
+                Fixture.TrumanSnomedDiagnosticReport,
+                Fixture.TrumanPatient); */
+
+            ValidateSearchEntryMode(bundle, ResourceType.Organization);
+        }
+
+        [Fact]
+        public async Task GivenAnIncludeSearchExpressionWithSimpleSearchAndCount_WhenSearched_ThenCorrectBundleShouldBeReturned()
+        {
+            string query = $"_tag={Fixture.Tag}&_revinclude=Location:managingOrganization&code=429858000&_count=1";
+
+            Bundle bundle = await Client.SearchAsync(ResourceType.Organization, query);
+
+            /* ValidateBundle(
+                bundle,
+                Fixture.SmithSnomedDiagnosticReport,
+                Fixture.SmithPatient); */
+
+            ValidateSearchEntryMode(bundle, ResourceType.Organization);
+
+            bundle = await Client.SearchAsync(bundle.NextLink.ToString());
+
+            ValidateBundle(
+                bundle,
+                Fixture.TrumanSnomedDiagnosticReport,
+                Fixture.TrumanPatient);
+
+            ValidateSearchEntryMode(bundle, ResourceType.DiagnosticReport);
+        }
+
+        /*
```

Comment:
Please avoid committing commented code. You can stash it or add it on a different branch for futures PRs.